qsh/__init__.py

```diff
@@ -190,23 +190,31 @@ class header:
     def __init__(self, filename=None, mode=None):
         self.fileobj = gzip.open(filename, mode)
 
-        signature = self.read(len(self.header.signature.encode("utf8")))
-
-        if self.header.signature != signature:
+        signature = self.read(len(QshFileHeader.signature.encode("utf8")))
+        if QshFileHeader.signature != signature:
             self.fileobj = io.open(filename, mode)
 
-            signature = self.read(len(self.header.signature.encode("utf8")))
+            signature = self.read(len(QshFileHeader.signature.encode("utf8")))
 
-            if self.header.signature != signature:
+            if QshFileHeader.signature != signature:
                 raise TypeError("Unsupported file format")
 
-        self.header.version       = self.read_byte()
-        self.header.application   = self.read_string()
-        self.header.comment       = self.read_string()
+        self.header = QshFileHeader()
+        self.header.version = self.read_byte()
+        self.header.application = self.read_string()
+        self.header.comment = self.read_string()
         self.last_created_at_milliseconds = to_milliseconds(self.read_datetime())
-        self.header.created_at    = to_datetime(self.last_created_at_milliseconds).replace(tzinfo=self.from_zone).astimezone(self.to_zone)
+        self.header.created_at = to_datetime(self.last_created_at_milliseconds).replace(tzinfo=self.from_zone).astimezone(self.to_zone)
         self.header.streams_count = self.read_byte()
 
+        # Last values for read_ord_log_data()
```

Comment:
Also I moved those mutable attributes from class to an instance to improve isolation between instances of `QshFile`